src/Rules.hs

```diff
@@ -90,3 +95,29 @@ buildRules = do
     Rules.Libffi.libffiRules
     packageRules
     Rules.Perl.perlScriptRules
+
+programsStage1Only :: [Package]
+programsStage1Only =
+  [ deriveConstants, genprimopcode, hp2ps, runGhc
```

Comment:
Not really, `needBuilder (Ghc CompileHs Stage1)` or `buildPackage Stage0 ghc` will only build:

```
deriveConstants genapply        genprimopcode   ghc-pkg         ghc-stage1
```

which is less then my previous comment about what the legacy system would build.